fml/platform/darwin/cf_utils.h

```diff
@@ -11,53 +11,99 @@
+///
+/// CFRef takes over ownership of the object it wraps and ensures that CFRetain
+/// and CFRelease are called as appropriate on creation, assignment, and
+/// disposal.
 template <class T>
 class CFRef {
  public:
+  /// Creates a new null CFRef.
   CFRef() : instance_(nullptr) {}
 
+  /// Takes over ownership of `instance`, which is expected to be already
+  /// retained.
+  ///
   // NOLINTNEXTLINE(google-explicit-constructor)
   CFRef(T instance) : instance_(instance) {}
 
+  /// Copy ctor: Creates a retained copy of the CoreFoundation object owned by
+  /// `other`.
   CFRef(const CFRef& other) : instance_(other.instance_) {
     if (instance_) {
       CFRetain(instance_);
     }
   }
 
+  /// Move ctor: Takes over ownership of the CoreFoundation object owned
+  /// by `other`. The object owned by `other` is set to null.
   CFRef(CFRef&& other) : instance_(other.instance_) {
     other.instance_ = nullptr;
   }
 
+  /// Takes over ownership of the CoreFoundation object owned by `other`.
   CFRef& operator=(CFRef&& other) {
     Reset(other.Release());
     return *this;
   }
 
+  /// Releases the underlying CoreFoundation object, if non-null.
   ~CFRef() {
-    if (instance_ != nullptr) {
+    if (instance_) {
       CFRelease(instance_);
     }
     instance_ = nullptr;
   }
 
+  /// Takes over ownership of `instance`, null by default. The object is
+  /// expected to be already retained if non-null.
+  ///
+  /// Releases the previous object, if non-null.
   void Reset(T instance = nullptr) {
-    if (instance_ != nullptr) {
+    if (instance_) {
       CFRelease(instance_);
     }
-
     instance_ = instance;
   }
 
+  /// Retains a shared copy of `instance`. The previous object is released if
+  /// non-null. Has no effect if `instance` is the currently-held object.
+  void Retain(T instance = nullptr) {
+    if (instance_ == instance) {
+      return;
+    }
+    if (instance) {
+      CFRetain(instance);
+    }
+    Reset(instance);
+  }
+
+  /// Returns and transfers ownership of the underlying CoreFoundation object
+  /// to the caller. The caller is responsible for calling `CFRelease` when done
+  /// with the object.
   [[nodiscard]] T Release() {
     auto instance = instance_;
     instance_ = nullptr;
     return instance;
   }
 
+  /// Returns the underlying CoreFoundation object. Ownership of the returned
+  /// object follows The Get Rule.
+  ///
+  /// See:
+  /// https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-SW1
+  T Get() const { return instance_; }
+
+  /// Returns the underlying CoreFoundation object. Ownership of the returned
+  /// object follows The Get Rule.
+  ///
+  /// See:
+  /// https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html#//apple_ref/doc/uid/20001148-SW1
   // NOLINTNEXTLINE(google-explicit-constructor)
```

Comment:
I would also be curious how often we rely on this implicit conversion, and if it's not too obnoxious to change, remove it and use `Get()` everywhere. This one probably isn't dangerous (although IIRC `scoped_nsobject` didn't have it, and there was a reason?), but implicit conversions when dealing with memory management make me nervous.

Reply:
The answer appears to be that we use this in a _lot_ of places, but generally agreed it would be more readable/obvious if we used `Get()`. I think that makes sense as a followup patch though.